src/elf.rs

```diff
@@ -1,18 +1,25 @@
+use crate::error::{GetSymbolsError, Result};
 use compact_symbol_table::CompactSymbolTable;
 use goblin::elf;
 use object::{ElfFile, Object, Uuid};
 use std::cmp;
-
 const UUID_SIZE: usize = 16;
 const PAGE_SIZE: usize = 4096;
 
-pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Option<CompactSymbolTable> {
-    let elf_file = ElfFile::parse(buffer).ok()?;
-    let elf_id = get_elf_id(&elf_file, buffer)?;
-    if format!("{:X}0", elf_id.simple()) != breakpad_id {
-        return None;
+pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Result<CompactSymbolTable> {
+    let elf_file = ElfFile::parse(buffer).or(Err(GetSymbolsError::InvalidInputError(
```

Comment:
Could you use `.map_err` in all the places where you're using `.or`? That will avoid creating the error object in the passing case.

Reply:
`.map_err` needs a function as a first parameter. Do you mean by `let elf_file = buffer.map_err(ElfFile::parse, Err(GetSymbolsError::InvalidInputError(
        "Could not parse ELF header",
    )))?;`?

Reply:
The function that you pass would be a lambda function that creates the error object. For example:

```rust
let elf_file = ElfFile::parse(buffer).map_err(|_| GetSymbolsError::InvalidInputError("Could not parse ELF header"))?;
```

or even

```rust
let elf_file = ElfFile::parse(buffer).map_err(|e| GetSymbolsError::ElfFileParseError(e))?;
```